src/OrchardCore.Modules/OrchardCore.Users/Views/Items/CustomUserSettingsDeploymentStep.Fields.Edit.cshtml

```diff
@@ -0,0 +1,65 @@
+@model CustomUserSettingsDeploymentStepViewModel
+
+@{
+    var settingsTypeNames = Model.SettingsTypeNames;
+    var allSettingsTypeNames = Model.AllSettingsTypeNames;
+}
+
+<h5>@T["Custom User Settings"]</h5>
+
+<div class="mb-3 mt-4 mb-5">
+    <div class="row">
+        <div class="col">
+            <div class="form-check">
+                <label class="form-check-label">
+                    <input class="form-check-input" asp-for="IncludeAll"
+                           data-reverseToggle=".cssel" />
+                    @T["Include all custom user settings."]
+                </label>
+            </div>
+        </div>
+    </div>
+</div>
+
+<div class="cssel mt-4 mb-5 @(Model.IncludeAll ? "collapse" : "show")">
+    <div class="row">
+        <div class="col-small">
+            <span class="hint">@T["The custom user settings to add as part of the plan."]</span>
+        </div>
+    </div>
+    <div class="row">
+        <div class="col-small">
+            <ul class="list-group">
+                @foreach (var settingsTypeName in allSettingsTypeNames)
+                {
+                    var checkd = settingsTypeNames?.Contains(settingsTypeName);
+
+                    <li class="list-group-item">
+                        <div class="form-check">
+                            <label class="form-check-label">
+                                <input class="form-check-input" type="checkbox" name="@Html.NameFor(m => m.SettingsTypeNames)" value="@settingsTypeName" checked="@checkd">
+                                @settingsTypeName
+                            </label>
+                        </div>
+                    </li>
+                }
+            </ul>
+        </div>
+    </div>
+</div>
+
+<script at="Foot">
+    window.addEventListener('DOMContentLoaded', function() {
+    var toggles = document.querySelectorAll('[data-reversetoggle]');
```

Comment:
I think `getElementsByClassName` will have better performance over querying element by data-attribute.

Reply:
But then we I need to add a new class to the checkbox input so it doesn't conflict with the other checkboxes.

Reply:
I'll give it a go